### ISSUE TITLE
Slightly confusing documentation around defaults in 1.3.0

### ISSUE THREAD

**Issue by jonhoo** [bincode-org/bincode#333]
I'm super happy to see 1.3.0 out — congratulations :tada:

I did, however, _immediately_ get bit by the new config stuff. Specifically, that `DefaultOptions` does not match the behavior of `bincode::serialize` and `bincode::deserialize`. This is because the free functions use
```rust
DefaultOptions::new()
    .with_fixint_encoding()
    .allow_trailing_bytes()
```
Presumably to maintain backwards compatibility.

I think it'd be useful to clarify this difference in behavior on `DefaultOptions`, `serialize`/`deserialize`, or all of the above. Currently, the docs say:

`deserialize`:
> Deserializes a slice of bytes into an instance of `T using the default configuration.

`serialize`:
> Serializes a serializable object into a `Vec` of bytes using the default configuration.

`DefaultOptions`:
> The default options for bincode serialization/deserialization.

Yet, as mentioned, these "defaults" do not match. This also applies to the other free-standing functions.

**Comment by ZoeyR** [comment]
Thanks for noticing this! I thought I had updated all the docs about config.

**Comment by ghost** [comment]
This has been corrected in #373
